### ISSUE TITLE
[Single post view]

### ISSUE THREAD

**Issue by dankeck** [LiveFastEatTrashRaccoon/RaccoonForFriendica#811]
**Describe the bug**
When viewing a post, 6 buttons (additional actions, reply, boost, upvote, downvote, and bookmark) do not get focus and cannot be activated, using Android's screen reader, TalkBack.

**To Reproduce**
- Start the TalkBack screen reader.
- Open Raccoon and view a feed.
- Double tap a post to view just that post.
- Attempt to navigate to the three-dot additional actions button, or to the bottom bar, either using [Explore by touch or Explore by swiping](https://support.google.com/accessibility/android/answer/6006598?hl=en). 

**Expected behavior**
The additional actions, reply, boost, upvote, downvote, and bookmark buttons get focus. Their names are read by TalkBack, and they can be activated by tapping twice.

**Screenshots**
![A post displayed in the Raccoon Android app, with an arrow pointing to the three dot icon to the right of the poster's name](https://github.com/user-attachments/assets/0c80401f-9235-449e-8dbf-5a05a68b3c47)

![The bottom bar of the Raccoon Android app, showing 5 icons: reply, boost, upvote, downvote, and bookmark](https://github.com/user-attachments/assets/a4fd653e-e33e-45c4-a9ff-5ea9d1b8b80e)

**Smartphone:**
 - Device: Pixel 5
 - OS: Android 14
 - Version 0.4.1-beta10 (84)

**Comment by AkesiSeli** [comment]
Post cards are merged into a single item, those buttons should be accessible as custom actions on the "root" element.

This behavior is intentional and post cards are specifically designed like this. I remember @pvagner explaining this is a best practice for accessibility/usability, this is why we changed the original implementation (resetting semantics of clickable internal item and adding custom actions on their outermost parent.

**Comment by pvagner** [comment]
Dear @dankeck,
As @AkesiSeli is explaining this is deliberate choice from my part. And since I was first person to suggest accessibility related improvements I had a luxury to being regarded as trustworthy in this regard.

When using talkback gestures to navigate through the UI, talkback is announcing there are actions available and it is also easy to choose the action by swiping up and down or down and up in a single gesture.

Frankly I am somewhat surprised you haven't discovered these accessibility actions on your own.
A classic example where these actions are also available is in the list of conversations in the gmail app, or on the home screen of most launchers i.e. pixel launcher, one ui launcher on samsung devices and similar.

The general idea is that when you are reading your social feed you should be able to navigate from one post to the next one by performing as few gestures as possible.

Now when you know there are these accessibility actions, can you please let us know how much do you like this design choice?

Thanks and greetings

Peter

**Comment by pvagner** [comment]
Dear @dankeck is it now working fine for you with talkback?
Are you using it your-self or are you helping out someone else?
Do you need further help related to accessibility of this app please?

**Comment by AkesiSeli** [comment]
Don't hesitate to report other issues (be it for accessibility or anything else), the app can be improved for sure.

**Comment by dankeck** [comment]
Peter and AkesiSeli, sorry for the delay in writing back. I really do appreciate that you pointed out accessibility actions to me. 

I'm a non-blind user who has been working the last few years to learn how to evaluate the accessibility of websites and software. Most of my efforts have been on websites, using the WCAG standards. But I'm trying to learn more about accessibility in mobile apps. My hope is that my favorite open source apps can continually be made more accessible to disabled users.

So, I'm not a native screen reader user and am far less proficient at using TalkBack and VoiceOver than native users.

You're correct to be surprised that I didn't know about the accessibility options. This is something I completely missed and should have known. Let me apologize for my ignorance. I'll blame myself for not talking more often to native screen reader users. After you told me about this Android feature, I've started using it and agree this is better than swiping through all the buttons. Now I feel slightly more prepared to submit accessibility bugs for other Android apps (but I still have plenty to learn). 

I know you'd probably like to have feedback from a blind user, or other native screen reader user. Unfortunately as I sighted person I can't provide authentic feedback. But thanks for offering to let me continue reporting any issues I find. Raccoon is a great app, and I appreciate all the work you've done so far.
